### PR TITLE
Improve fixtures

### PR DESCRIPTION
**Pull request by wbloszyk**
This PR change templates in fixtures from string to render by twig templates. This will help in generating layouts/templates for other css framework (like bootstrap4).

### CODE REVIEW

File: src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadOrderData.php [hunk 1]
```diff
@@ -24,7 +24,7 @@
 use Doctrine\Common\Collections\ArrayCollection;
 use Doctrine\Common\DataFixtures\AbstractFixture;
```
Comment (jordisala1991) [inline src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadOrderData.php:25]: I think those other classes were moved too, maybe it could be good doing a separate PR with all this changes?

**Comment by wbloszyk** [comment]
RTM

File: src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadPageData.php [hunk 2]
```diff
@@ -15,28 +15,79 @@
...
+     */
+    protected $createSubsite;
+
+    public function __construct(
+        Generator $fakeGenerator,
+        BlockInteractorInterface $blockInteractor,
+        BlockManagerInterface $blockManager,
+        SiteManagerInterface $siteManager,
+        PageManagerInterface $pageManager,
+        Environment $environment,
```
Comment (jordisala1991) [inline src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadPageData.php:71]: $twig

File: src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadPageData.php [hunk 3]
```diff
@@ -15,28 +15,79 @@
...
-    public function setContainer(ContainerInterface $container = null)
+    public function getOrder()
     {
-        $this->container = $container;
+        return 8;
     }
 
     public function load(ObjectManager $manager)
     {
+        $twig = $this->twig;
```
Comment (jordisala1991) [inline src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadPageData.php:90]: do not create extra var

File: src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadPageData.php [hunk 4]
```diff
@@ -226,9 +261,10 @@ public function createTermsPage(SiteInterface $site)
 
     public function createHomePage(SiteInterface $site)
     {
-        $pageManager = $this->getPageManager();
-        $blockManager = $this->getBlockManager();
-        $blockInteractor = $this->getBlockInteractor();
+        $twig = $this->twig;
+        $pageManager = $this->pageManager;
+        $blockManager = $this->blockManager;
+        $blockInteractor = $this->blockInteractor;
```
Comment (jordisala1991) [inline src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadPageData.php:267]: we dont need to create extra vars

File: src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadPageData.php [hunk 5]
```diff
@@ -334,8 +356,8 @@ public function createHomePage(SiteInterface $site)
             'enabled' => true,
             'page' => $homepage,
             'code' => 'bottom_embed',
-        ], static function ($container) {
-            $container->setSetting('layout', '<div class="col-sm-6">{{ CONTENT }}</div>');
+        ], static function ($container) use ($twig) {
```
Comment (jordisala1991) [inline src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadPageData.php:359]: removing static will let you use $this inside

File: src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadProductData.php [hunk 6]
```diff
@@ -20,44 +20,77 @@
...
+
+    public function getOrder()
+    {
+        return 7;
+    }
 
     public function load(ObjectManager $manager)
     {
-        $productPool = $this->getProductPool();
+        $productPool = $this->productPool;
```
Comment (jordisala1991) [inline src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadProductData.php:82]: same, do not create extra vars

File: src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadProductData.php [hunk 7]
```diff
@@ -1129,7 +1130,7 @@ protected function addMediaToProduct($mediaFilename, $name, $description, Produc
      */
     protected function createMedia($mediaFilename, $name, $description, $author = null, $copyright = null, $categoryReference = 'products_category')
     {
-        $mediaManager = $this->getMediaManager();
+        $mediaManager = $this->mediaManager;
```
Comment (jordisala1991) [inline src/Sonata/Bundle/DemoBundle/DataFixtures/ORM/LoadProductData.php:1133]: This var should not be needed

**Comment by wbloszyk** [comment]
I will do another RP for this fixtures. Some things can still be improved, for example, transfer data to arrays, create dedicated methods instead of loading hundreds or thousands of lines of code.

**Comment by jordisala1991** [comment]
Thank you @wbloszyk
